File: deleteme/JSLambdaCode.js
"use strict";
class JSLambdaCode {
	constructor() {
		
	}
	static __isTransient() {
		return false;
	}
	static __getClass() {
		return "ca.frar.jjjrmi.jsbuilder.code.JSLambdaCode";
	}
	static __isEnum() {
		return false;
	}
	callsLambda1() {
		this.takesLambda((e)=>e.toString());
	}
	callsLambda2() {
		this.takesLambda((e)=>{
			e = e.concat(" i really am");
			e.toString();
		});
	}
	takesLambda(consumer) {
		consumer.accept("ima string");
	}
};

if (typeof module !== "undefined") module.exports = JSLambdaCode;

File: src/test/java/ca/frar/jjjrmi/testclasses/JSLambdaCode.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ca.frar.jjjrmi.testclasses;

import ca.frar.jjjrmi.annotations.NativeJS;
import ca.frar.jjjrmi.socket.JJJObject;
import java.util.function.Consumer;

/**
 *
 * @author Ed Armstrong
 */
public class JSLambdaCode extends JJJObject{
    
    @NativeJS
    public void takesLambda(Consumer<String> consumer){
        consumer.accept("ima string");
    }
    
    @NativeJS
    public void callsLambda1(){
        this.takesLambda((e)->e.toString());
    }

    @NativeJS
    public void callsLambda2(){
        this.takesLambda((e)->{
            e = e.concat(" i really am");
            e.toString();
        });
    }
    
}
